src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,381 @@
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsUnifiedCode;
+import org.prebid.server.analytics.reporter.greenbids.model.MediaTypes;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.EncodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.log.Logger;
+import org.prebid.server.log.LoggerFactory;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.NonBid;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.SeatNonBid;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.version.PrebidVersionProvider;
+import org.prebid.server.vertx.httpclient.HttpClient;
+import org.prebid.server.vertx.httpclient.model.HttpClientResponse;
+
+import java.time.Clock;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GreenbidsAnalyticsReporter implements AnalyticsReporter {
+
+    private static final String PUBLISHER_ID_DYNAMIC_PARAM = "pbuid";
+    private static final String GREENBIDS_SAMPLING_DYNAMIC_PARAM = "greenbidsSampling";
+    private static final String BID_REQUEST_ANALYTICS_EXTENSION_NAME = "greenbids";
+    private static final int RANGE_16_BIT_INTEGER_DIVISION_BASIS = 0x10000;
+    private static final Logger logger = LoggerFactory.getLogger(GreenbidsAnalyticsReporter.class);
+    private final GreenbidsAnalyticsProperties greenbidsAnalyticsProperties;
+    private final JacksonMapper jacksonMapper;
+    private final HttpClient httpClient;
+    private final Clock clock;
+    private final PrebidVersionProvider prebidVersionProvider;
+
+    public GreenbidsAnalyticsReporter(
+            GreenbidsAnalyticsProperties greenbidsAnalyticsProperties,
+            JacksonMapper jacksonMapper,
+            HttpClient httpClient,
+            Clock clock,
+            PrebidVersionProvider prebidVersionProvider) {
+        this.greenbidsAnalyticsProperties = Objects.requireNonNull(greenbidsAnalyticsProperties);
+        this.httpClient = Objects.requireNonNull(httpClient);
+        this.clock = Objects.requireNonNull(clock);
+        this.prebidVersionProvider = Objects.requireNonNull(prebidVersionProvider);
+
+        final ObjectMapper mapper = Objects.requireNonNull(jacksonMapper).mapper()
+                .setPropertyNamingStrategy(PropertyNamingStrategy.LOWER_CAMEL_CASE);
```

Comment:
I don't the problem in adding @JsonProperty annotation, but if you really want your own mapper then please create it separately without changing the existing one, in that way you won't be dependent on the changes that might be done on the higher level that could bring backward compatibility problems in future

Reply:
Thanks, I think I can just create a custom JacksonMapper specifically for our module